extensions/access_log_policy/plugin.cc

```diff
@@ -0,0 +1,143 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "extensions/access_log_policy/plugin.h"
+
+#include <google/protobuf/io/zero_copy_stream_impl_lite.h>
+
+#include "absl/strings/str_cat.h"
+#include "absl/strings/str_split.h"
+#include "extensions/common/istio_dimensions.h"
+#include "extensions/common/node_info.pb.h"
+#include "google/protobuf/util/json_util.h"
+
+#ifndef NULL_PLUGIN
+
+#include "base64.h"
+
+#else
+
+#include "common/common/base64.h"
+namespace Envoy {
+namespace Extensions {
+namespace Wasm {
+namespace AccessLogPolicy {
+namespace Plugin {
+using namespace ::Envoy::Extensions::Common::Wasm::Null::Plugin;
+using NullPluginRootRegistry =
+    ::Envoy::Extensions::Common::Wasm::Null::NullPluginRootRegistry;
+using google::protobuf::util::JsonParseOptions;
+using google::protobuf::util::Status;
+
+NULL_PLUGIN_ROOT_REGISTRY;
+
+#endif
+
+namespace {
+
+bool setFilterStateValue(bool log) {
+  auto r = setFilterStateStringValue(::Wasm::Common::kAccessLogPolicyKey,
+                                     log ? "yes" : "no");
+  if (r != WasmResult::Ok) {
+    logWarn(toString(r));
+    return false;
+  }
+  return true;
+}
+
+}  // namespace
+
+constexpr long int kDefaultLogWindowDurationNanoseconds =
+    43200000000000;  // 12h
+
+constexpr StringView kClientIp = "client_ip";
+constexpr StringView kPrincipal = "principal";
+constexpr StringView kDestination = "destination";
+constexpr StringView kAddress = "address";
+constexpr StringView kConnection = "connection";
+constexpr StringView kUriSanPeerCertificate = "uri_san_peer_certificate";
+
+static RegisterContextFactory register_AccessLogPolicy(
+    CONTEXT_FACTORY(PluginContext), ROOT_FACTORY(PluginRootContext));
+
+bool PluginRootContext::onConfigure(size_t) {
+  if (::Wasm::Common::TrafficDirection::Outbound ==
+      ::Wasm::Common::getTrafficDirection()) {
+    throw EnvoyException("ASM Acess Logging Policy is an inbound filter only.");
+  }
+  WasmDataPtr configuration = getConfiguration();
+  JsonParseOptions json_options;
+  Status status =
+      JsonStringToMessage(configuration->toString(), &config_, json_options);
+  if (status != Status::OK) {
+    logWarn("Cannot parse Stackdriver plugin configuration JSON string " +
+            configuration->toString() + ", " + status.message().ToString());
+    return false;
+  }
+
+  if (config_.has_log_window_duration()) {
+    log_time_duration_nanos_ =
+        ::google::protobuf::util::TimeUtil::DurationToNanoseconds(
+            config_.log_window_duration());
+  } else {
+    log_time_duration_nanos_ = kDefaultLogWindowDurationNanoseconds;
+  }
+
+  // We currently cache the key based on client_ip and principal only.
+  IstioDimensions::InitIstioDimensions({kClientIp, kPrincipal});
```

Comment:
I can't find this symbol `InitIstioDimensions`. Is this some macro? Is this a singleton? Doesn't it interfere with stats extension?

Reply:
It's a static function within istiodimensions.h